src/adapter/src/coord/sequencer.rs

```diff
@@ -2133,23 +2134,17 @@ impl<S: Append + 'static> Coordinator<S> {
         }
 
         let source_ids = source.depends_on();
-        let timeline = self.validate_timeline(source_ids.clone())?;
-        // Queries are independent of the logical timestamp iff there are no referenced
-        // sources or indexes and there is no reference to `mz_now()`.
-        let timestamp_independent = source_ids.is_empty() && !source.contains_temporal();
+        let mut timeline_context = self.validate_timeline_context(source_ids.clone())?;
+        if matches!(timeline_context, TimelineContext::TimestampIndependent)
+            && source.contains_temporal()
```

Comment:
My read of `get_timeline_contexts` is it already does this. Do we need this if block?

Reply:
Unfortunately, `source_ids` does not contain any functions. So queries that use `mz_now()` directly won't be caught by `let mut timeline_context = self.validate_timeline_context(source_ids.clone())?;`